lib/matplotlib/axes/_axes.py

```diff
@@ -838,6 +838,75 @@ def axvline(self, x=0, ymin=0, ymax=1, **kwargs):
         self.autoscale_view(scalex=scalex, scaley=False)
         return l
 
+    @docstring.dedent_interpd
+    def axline(self, slope, intercept, **kwargs):
+        """
+        Add an infinite straight line across an axis.
+
+        Parameters
+        ----------
+        slope : scalar
+            Slope of the line.
+
+        intercept : scalar
+            Intercept of the line with the y-axis.
+
+        Returns
+        -------
+        :class:`~matplotlib.lines.Line2D`
+
+        Other Parameters
+        ----------------
+        Valid kwargs are :class:`~matplotlib.lines.Line2D` properties,
+        with the exception of 'transform':
+
+        %(_Line2D_docstr)s
+
+        Examples
+        --------
+        * Draw a thick red line with slope 1 and y-intercept 0::
+
+            >>> axline(1, 0, linewidth=4, color='r')
+
+        * Draw a default line with slope 1 and y-intercept 1::
+
+            >>> axline(1, 1)
+
+        * Draw a default line with slope 5 and y-intercept 0::
+
+            >>> axline(5, 0)
+
+        See Also
+        --------
+        axhline : for horizontal lines
+        axvline : for vertical lines
+
+        Notes
+        -----
+        Currently this method does not work properly with log scaled axes.
```

Comment:
Would rather write "only works with linear axes" (there's log, symlog, and whatnot).